### PR TITLE
Inductive Infomax

### PR DESCRIPTION
**Pull request by harrygcoppock**
Hi,

First of all, LOVE the pytorch geometric library.

### [Summary of contribution]
The pytorch geometric library has an implementation of transductive Infomax (see examples/infomax.py) however there is no implementation of Inductive Infomax. I have added an example of an implementation of inductive infomax for the reddit dataset. The relevant paper is: "Deep Graph Infomax" <https://arxiv.org/abs/1809.10341>

### [Performance]
The paper reports an accuracy of 94%, here we achieve an accuracy of 93.55% 

### [Tests and formatting]
All formatting tests (flake8 and pycodestyle) pass with no errors and `python setup.py test` returns no errors:
Screenshot 2020-11-24 at 10 21 19

I couldn't see any tests for examples/ so didn't create any for Inductive_Infomax.py - is this okay?

### [General Comments]
- I only test classification accuracy at the end of training as the LogisticRegression func takes 20x the time it takes to train just one epoch.

Happy to implement any changes you propose.

### CODE REVIEW

**Comment by rusty1s** [comment]
Thank you very much. This looks excellent. I will have a closer look ASAP.

**Comment by rusty1s** [comment]
Thanks once again :)

**Comment by harrygcoppock** [comment]
No problem! I enjoyed using the great Library. That you for cleaning my code - still new to all of this. (1st year PhD student - thought that I should get my first contribution to open source code out of the way ahah)

**Comment by rusty1s** [comment]
Your code was awesome :) I only did some small changes (which are debatable :))

**Comment by rusty1s** [comment]
BTW: I reduced the number of epochs since it reaches 93.5 after only a few epochs. I did not see a reason to train that long (although the paper reports to use 150 epochs).

**Comment by harrygcoppock** [comment]
Haha thank you that's great to hear! Yeah I did see that - was going to leave that decision to you.
